chainer/training/extensions/variable_statistics_plot.py

```diff
@@ -0,0 +1,253 @@
+            This is distinct from the trigger of this extension
+            itself. If it is a tuple in the form ``<int>, 'epoch'`` or
+            ``<int>, 'iteration'``, it is passed to :class:`IntervalTrigger`.
+        file_name (str): Name of the plot file under the output directory.
+        figsize (tuple of int): Matlotlib ``figsize`` argument that specifies
+            the size of the output image.
+        marker (str): Matplotlib ``marker`` argument that specified the marker
+            style of the plots.
+        grid (bool): Matplotlib ``grid`` argument that specifies whether grids
+            are rendered in in the plots or not.
+    """
+
+    def __init__(self, targets, max_sample_size=1000,
+                 report_data=True, report_grad=True,
+                 trigger=(1, 'epoch'), file_name='statistics.png',
+                 figsize=None, marker=None, grid=True):
+
+        if file_name is None:
+            raise ValueError('Missing output file name of statstics plot')
+
+        self._vars = _unpack_variables(targets)
+        if len(self._vars) == 0:
+            raise ValueError(
+                'Need at least one variables for which to collect statistics.'
+                '\nActual: 0 <= 0')
+
+        self._report_data = report_data
+        self._report_grad = report_grad
+        self._trigger = trigger_module.get_trigger(trigger)
+
+        self._file_name = file_name
+        self._figsize = figsize
+        self._marker = marker
+        self._grid = grid
+
+        self._keys = []
+        if report_data:
+            self._keys.append('data')
+        if report_grad:
+            self._keys.append('grad')
+
+        self._statistician = Statistician()
+        self._data_shape = (len(self._keys), self._statistician.data_size)
+        self._samples = Reservoir(max_sample_size, data_shape=self._data_shape)
+
+    @staticmethod
+    def available():
+        _check_available()
+        return _available
+
+    def __call__(self, trainer):
+        if not _available:
+            # This extension does nothing if matplotlib is not available
+            return
+
+        xp = cuda.get_array_module(self._vars[0].data)
+        stats = xp.zeros(self._data_shape, dtype=xp.float32)
+        for i, k in enumerate(self._keys):
+            xs = []
+            for var in self._vars:
+                x = getattr(var, k, None)
+                if x is not None:
+                    xs.append(x.ravel())
+            if len(xs) > 0:
+                stats[i] = self._statistician(
+                    xp.concatenate(xs, axis=0), axis=0, xp=xp)
+        if xp != numpy:
+            stats = cuda.to_cpu(stats)
+        self._samples.add(stats, idx=trainer.updater.iteration)
+
+        if self._trigger(trainer):
+            file_path = os.path.join(trainer.out, self._file_name)
+            self.save_plot(file_path)
+
+    def save_plot(self, file_path):
+        nrows = 2  # a mean, std plot and a percentiles plot
```

Comment:
Is it correct? `self._statician.datasize`?

Reply:
It is actually. The first row contains mean and std plots for `data`, and `grad`, and the second row contains percentile plots. Do you think this should be changed?